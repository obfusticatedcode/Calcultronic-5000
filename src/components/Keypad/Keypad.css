.Keypad__container {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  height: var(--keypad-height);
  padding: 2%;
  width: var(--keypad-width);
}

.Keypad__numbersContainer {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  height: var(--number-height);
  width: var(--number-width);
}

.Keypad__operatorsContainer {
  display: flex;
  flex-direction: column;
  height: 80%;
  width: 25%;
}

.Keypad__submitContainer {
  height: 20%;
  width: 100%;
  display: flex;
  align-items: center;
  justify-content: center;
  background-color: var(--submit-key-color);
}
