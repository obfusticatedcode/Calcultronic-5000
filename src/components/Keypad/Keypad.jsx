import React from "react";
import PropTypes from "prop-types";
import Key from "../Key/Key";

import "./Keypad.css";

const Keypad = ({
  callOperator,
  numbers,
  operators,
  setOperator,
  updateDisplay
}) => {
  const numberKeys = numbers.map(number => (
    <Key
      key={number}
      keyAction={updateDisplay}
      keyType="number-key"
      keyValue={number}
    />
  ));
  const operatorKeys = operators.map(operator => (
    <Key
      key={operator}
      keyAction={setOperator}
      keyType="operator-key"
      keyValue={operator}
    />
  ));
  return (
    <div className="Keypad__container">
      <div className="Keypad__numbersContainer">{numberKeys}</div>
      <div className="Keypad__operatorsContainer">{operatorKeys}</div>
      <div className="Keypad__submitContainer">
        <Key keyAction={callOperator} keyType="submit-key" keyValue="=" />
      </div>
    </div>
  );
};

Keypad.propTypes = {
  callOperator: PropTypes.func.isRequired,
  numbers: PropTypes.array.isRequired,
  operators: PropTypes.array.isRequired,
  setOperator: PropTypes.func.isRequired,
  updateDisplay: PropTypes.func.isRequired
};
export default Keypad;
