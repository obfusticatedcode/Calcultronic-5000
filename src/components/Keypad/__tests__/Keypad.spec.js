import React from "react";
import { mount, shallow } from "enzyme";
import Keypad from "../Keypad";

const callOperator = jest.fn();
const setOperator = jest.fn();
const updateDisplay = jest.fn();
const defaultProps = {
  callOperator,
  setOperator,
  updateDisplay,
  numbers: [],
  operators: []
};
const setup = (props = {}) => shallow(<Keypad {...defaultProps} {...props} />);
const mountedSetup = (props = {}) =>
  mount(<Keypad {...defaultProps} {...props} />);

describe("<Keypad></Keypad>", () => {
  let wrapper = null;
  beforeEach(() => (wrapper = setup({})));
  it("should render 4 <div></div> s", () => {
    expect(wrapper.find("div").length).toEqual(4);
  });

  it("should render an instance of the <Key/> for each index of numbers, operators, and the submit Key", () => {
    const numbers = ["0", "1"];
    const operators = ["+", "-"];
    const submit = 1;
    const keyTotal = numbers.length + operators.length + submit;
    wrapper.setProps({ numbers, operators });
    expect(wrapper.find("Key").length).toEqual(keyTotal);
  });
});

describe("mounted <Keypad/>", () => {
  let wrapper = null;
  beforeEach(() => {
    wrapper = mountedSetup({});
  });
  it("should render the values of numbers", () => {
    wrapper.setProps({ numbers: ["0", "1", "2"] });
    expect(wrapper.find(".Keypad__numbersContainer").text()).toEqual("012");
  });

  it("should render the values of operators", () => {
    wrapper.setProps({ operators: ["+", "-", "*", "/"] });
    expect(wrapper.find(".Keypad__operatorsContainer").text()).toEqual("+-*/");
  });

  it("should render correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
