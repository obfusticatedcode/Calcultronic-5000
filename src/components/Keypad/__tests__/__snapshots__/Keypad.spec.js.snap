// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mounted <Keypad/> should render correctly 1`] = `
ReactWrapper {
  Symbol(enzyme.__unrendered__): <Keypad
    callOperator={[MockFunction]}
    numbers={Array []}
    operators={Array []}
    setOperator={[MockFunction]}
    updateDisplay={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "callOperator": [MockFunction],
      "numbers": Array [],
      "operators": Array [],
      "setOperator": [MockFunction],
      "updateDisplay": [MockFunction],
    },
    "ref": null,
    "rendered": Object {
      "instance": <div
        class="Keypad__container"
      >
        <div
          class="Keypad__numbersContainer"
        />
        <div
          class="Keypad__operatorsContainer"
        />
        <div
          class="Keypad__submitContainer"
        >
          <div
            class="Key__container"
          >
            <p
              class="Key__value"
            >
              =
            </p>
          </div>
        </div>
      </div>,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="Keypad__numbersContainer"
          />,
          <div
            className="Keypad__operatorsContainer"
          />,
          <div
            className="Keypad__submitContainer"
          >
            <Key
              keyAction={[MockFunction]}
              keyType="submit-key"
              keyValue="="
            />
          </div>,
        ],
        "className": "Keypad__container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": <div
            class="Keypad__numbersContainer"
          />,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [],
            "className": "Keypad__numbersContainer",
          },
          "ref": null,
          "rendered": Array [
            Array [],
          ],
          "type": "div",
        },
        Object {
          "instance": <div
            class="Keypad__operatorsContainer"
          />,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [],
            "className": "Keypad__operatorsContainer",
          },
          "ref": null,
          "rendered": Array [
            Array [],
          ],
          "type": "div",
        },
        Object {
          "instance": <div
            class="Keypad__submitContainer"
          >
            <div
              class="Key__container"
            >
              <p
                class="Key__value"
              >
                =
              </p>
            </div>
          </div>,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Key
              keyAction={[MockFunction]}
              keyType="submit-key"
              keyValue="="
            />,
            "className": "Keypad__submitContainer",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "keyAction": [MockFunction],
                "keyType": "submit-key",
                "keyValue": "=",
              },
              "ref": null,
              "rendered": Object {
                "instance": <div
                  class="Key__container"
                >
                  <p
                    class="Key__value"
                  >
                    =
                  </p>
                </div>,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <p
                    className="Key__value"
                  >
                    =
                  </p>,
                  "className": "Key__container",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": <p
                      class="Key__value"
                    >
                      =
                    </p>,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "=",
                      "className": "Key__value",
                    },
                    "ref": null,
                    "rendered": Array [
                      "=",
                    ],
                    "type": "p",
                  },
                ],
                "type": "div",
              },
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "callOperator": [MockFunction],
        "numbers": Array [],
        "operators": Array [],
        "setOperator": [MockFunction],
        "updateDisplay": [MockFunction],
      },
      "ref": null,
      "rendered": Object {
        "instance": <div
          class="Keypad__container"
        >
          <div
            class="Keypad__numbersContainer"
          />
          <div
            class="Keypad__operatorsContainer"
          />
          <div
            class="Keypad__submitContainer"
          >
            <div
              class="Key__container"
            >
              <p
                class="Key__value"
              >
                =
              </p>
            </div>
          </div>
        </div>,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="Keypad__numbersContainer"
            />,
            <div
              className="Keypad__operatorsContainer"
            />,
            <div
              className="Keypad__submitContainer"
            >
              <Key
                keyAction={[MockFunction]}
                keyType="submit-key"
                keyValue="="
              />
            </div>,
          ],
          "className": "Keypad__container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": <div
              class="Keypad__numbersContainer"
            />,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [],
              "className": "Keypad__numbersContainer",
            },
            "ref": null,
            "rendered": Array [
              Array [],
            ],
            "type": "div",
          },
          Object {
            "instance": <div
              class="Keypad__operatorsContainer"
            />,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [],
              "className": "Keypad__operatorsContainer",
            },
            "ref": null,
            "rendered": Array [
              Array [],
            ],
            "type": "div",
          },
          Object {
            "instance": <div
              class="Keypad__submitContainer"
            >
              <div
                class="Key__container"
              >
                <p
                  class="Key__value"
                >
                  =
                </p>
              </div>
            </div>,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Key
                keyAction={[MockFunction]}
                keyType="submit-key"
                keyValue="="
              />,
              "className": "Keypad__submitContainer",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "keyAction": [MockFunction],
                  "keyType": "submit-key",
                  "keyValue": "=",
                },
                "ref": null,
                "rendered": Object {
                  "instance": <div
                    class="Key__container"
                  >
                    <p
                      class="Key__value"
                    >
                      =
                    </p>
                  </div>,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <p
                      className="Key__value"
                    >
                      =
                    </p>,
                    "className": "Key__container",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": <p
                        class="Key__value"
                      >
                        =
                      </p>,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "=",
                        "className": "Key__value",
                      },
                      "ref": null,
                      "rendered": Array [
                        "=",
                      ],
                      "type": "p",
                    },
                  ],
                  "type": "div",
                },
                "type": [Function],
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
